sunpy/instr/aia/response.py

```diff
@@ -0,0 +1,346 @@
+                'ccd_contamination'] * var['quantum_efficiency_ccd']
+
+            self.eff_area = eff_area
+        else:
+            # Returns effective area at the position of the center wavelength for the channel specified
+
+            # find the index in arrays from ssw for desired wavelength
+            for n, value in enumerate(self.get_wavelength_range(channel)):
+                if channel < value < channel + 1:
+                    # print(n,  value)
+                    index = n
+                    break
+
+            # import instrument properties
+            reflectance = var['primary_mirror_reflectance'][index] * var['secondary_mirror_reflectance'][index]
+            transmission_efficiency = var['focal_plane_filter_efficiency'][index] * var['entire_filter_efficiency'][index]
+
+            # equation calculation :
+            eff_area = (var['geometric_area_ccd'] * u.cm ** 2) * reflectance * transmission_efficiency * var[
+                'ccd_contamination'][index] * var['quantum_efficiency_ccd'][index]
+            self.eff_area = eff_area
+
+            if print_comparison_ratio:
+                # used for testing - move to test_aia?
+                compare_eff_area = [0.312, 1.172, 2.881, 1.188, 1.206, 0.063, 0.045, 0.0192, 0.0389][
+                    self.channel_list.index(channel)]
+                print('eff_area: ', compare_eff_area, eff_area, 'ratio of comparison: ', eff_area / compare_eff_area)
+
+        return self.eff_area
+
+    def get_wavelength_response(self, channel, use_response_table2=False, use_calc_effarea_table2_gain=False,
+                                use_genx_values=False):
+        """
+
+        Describes the instrument (AIA) response per wavelength by calculating effective area vs wavelength of the strongest emission lines present in the solar feature.
+
+        # formula from paper
+        [R(wavelength, t) = A eff (wavelength, t)G(wavelength)]
+
+
+        Parameters
+        ----------
+        :keyword
+
+
+        Returns
+        -------
+        :return: float, array describing the response per wavelength of effective area (wavelength response)
+                 count rate for spectrum?  : units cm^2 DN phot^-1
+
+
+        NOTE:
+
+        The energy of a photon E = hf = hc / lambda . Expressing hc in units of eV = 12398.4953
+
+        """
+
+        var = self.get_channel_data(channel)
+
+        # conversions
+        dn_per_photon = u.count / u.photon
+        ev_per_angstrom = (12398.4953 * u.eV * u.angstrom)
+        electron_per_ev = (1 * u.electron) / (3.98 * u.eV)
+
+
+        # gain values from Boerner paper
+        gain_table2 = {94: 2.128, 131: 1.523, 171: 1.168, 193: 1.024, 211: 0.946, 304: 0.658, 335: 0.596, 1600: 0.125,
+                       1700: 0.118}
+        ccd_gain = {94: 18.3, 131: 17.6, 171: 17.7, 193: 18.3, 211: 18.3, 304: 18.3, 335: 17.6, 1600: 0.125,
```

Comment:
Don't need to hardcode this. Available from the `elecperdn` field in the genx files.